src/sql/src/plan/statement/ddl/connection.rs

```diff
@@ -0,0 +1,490 @@
+            SslCertificateAuthority,
+            SslKey,
+            Url,
+            User,
+        ],
+        CreateConnectionType::Kafka => &[
+            Broker,
+            Brokers,
+            ProgressTopic,
+            SshTunnel,
+            SslKey,
+            SslCertificate,
+            SslCertificateAuthority,
+            SaslMechanisms,
+            SaslUsername,
+            SaslPassword,
+        ],
+        CreateConnectionType::Postgres => &[
+            AwsPrivatelink,
+            Database,
+            Host,
+            Password,
+            Port,
+            SshTunnel,
+            SslCertificate,
+            SslCertificateAuthority,
+            SslKey,
+            SslMode,
+            User,
+        ],
+        CreateConnectionType::Ssh => &[Host, Port, User],
+    };
+
+    for o in permitted_options {
+        options.remove(o);
+    }
+
+    if !options.is_empty() {
+        sql_bail!(
+            "{} connections do not support {} values",
+            t,
+            options.iter().join(", ")
+        )
+    }
+
+    Ok(())
+}
+
+impl ConnectionOptionExtracted {
+    pub(super) fn ensure_only_valid_options(
+        &self,
+        t: CreateConnectionType,
+    ) -> Result<(), PlanError> {
+        validate_options_per_connection_type(t, self.seen.clone())
+    }
+
+    pub fn try_into_connection(
+        self,
+        scx: &StatementContext,
+        connection_type: CreateConnectionType,
+    ) -> Result<Connection<ReferencedConnection>, PlanError> {
+        self.ensure_only_valid_options(connection_type)?;
+
+        let connection: Connection<ReferencedConnection> = match connection_type {
+            CreateConnectionType::Aws => {
+                Connection::Aws(AwsConfig {
+                    credentials: AwsCredentials {
+                        access_key_id: self
+                            .access_key_id
+                            .ok_or_else(|| sql_err!("ACCESS KEY ID option is required"))?,
+                        secret_access_key: self
+                            .secret_access_key
+                            .ok_or_else(|| sql_err!("SECRET ACCESS KEY option is required"))?
+                            .into(),
+                        session_token: self.token,
+                    },
+                    endpoint: match self.endpoint {
+                        // TODO(benesch): this should not treat an empty endpoint as equivalent to a `NULL`
+                        // endpoint, but making that change now would break testdrive. AWS connections are
+                        // all behind feature flags mode right now, so no particular urgency to correct
+                        // this.
+                        Some(endpoint) if !endpoint.is_empty() => Some(endpoint),
+                        _ => None,
+                    },
+                    region: self.region,
+                    role: self.role_arn.map(|arn| AwsAssumeRole { arn }),
+                })
```

Comment:
Untested according to coverage data

Reply:
This is just code movement in per, so will excuse myself from this. AWS connections aren't currently used anywhere.